### PR TITLE
drop use of artifactory maven repo

### PR DESCRIPTION
**Pull request by jakubgs**
We really need to get rid of Artifactory. I wonder fi this would be enough to drop dependency on it...

### CODE REVIEW

**Review by vitvly — APPROVED** [review]

**Comment by pedropombeiro** [comment]
@jakubgs I'm actually working on a related issue (remove the need to build status-go every time on Desktop builds), and I'm starting by generating the required static library from the Go side: https://github.com/status-im/status-go/issues/1190

File: ci/mobile.groovy [hunk 1]
```diff
@@ -51,6 +51,12 @@ def prep(type = 'nightly') {
   sh 'cp patches/js_realm.hpp node_modules/realm/src/js_realm.hpp'
   /* install ruby dependencies */
   sh 'bundle install --quiet'
+  /* download status-go for mvn */
+  def pomDir = 'modules/react-native-status/ios/RCTStatus'
+  def repo = "${pwd}/${pomDir}/repo/status-im/status-go-ios/1.15.0"
```
Comment (pedropombeiro) [inline ci/mobile.groovy]: `1.15.0`: you mean `0.15.0` right?

also, ideally there would be a single line needing to be updated with this by the `scripts/update-status-go`.

File: Makefile [hunk 2]
```diff
@@ -3,6 +3,8 @@
 help: ##@other Show this help
 	@perl -e '$(HELP_FUN)' $(MAKEFILE_LIST)
 
+GO_VERSION = 0.15.0
```
Comment (pedropombeiro) [inline Makefile]: Maybe we can read this from the platform-specific file (.pom file for iOS, `modules/react-native-status/android/build.gradle` for Android)?

**Review by pedropombeiro — CHANGES_REQUESTED** [review]
Looking good! Need to account for remaining platforms, but once I have static libraries being built for Desktop, it's going to be golden and we'll have shorter CI builds.

File: Makefile (hunk 2 above)
Comment (jakubgs) [reply Makefile]: Yeah, I was thinking about that. Kinda flimsy, but better than hardcoding it.

File: ci/mobile.groovy (hunk 1 above)
Comment (jakubgs) [reply ci/mobile.groovy]: That's true, consider this a work in progress. I still have a long way to go to make this work with `pom.xml`. It's so painful.

File: modules/react-native-status/android/build.gradle [hunk 3]
```diff
@@ -16,7 +16,9 @@ dependencies {
     implementation 'com.instabug.library:instabug:3+'
     implementation 'status-im:function:0.0.1'
```
Comment (pedropombeiro) [inline modules/react-native-status/android/build.gradle:22]: @jakubgs Where would we get this `status-im:function` module from? It is currently in artifactory.

**Comment by jakubgs** [comment]
Turns out the `pom.xml` file is useless beyond downloading the iOS status-go file and extracting it from a Maven respository. Since we are dropping Artifactry we can also drop our dependency on Maven, so I removed `pom.xml` and I just do the same in the `Makefile`.

File: modules/react-native-status/android/build.gradle (hunk 3 above)
Comment (jakubgs) [reply modules/react-native-status/android/build.gradle:22]: If you look closer you can see that I've added `android/app/libs/function-0.0.1.jar` to the repo itself. It's a tiny file.

File: Makefile (hunk 2 above)
Comment (jakubgs) [reply Makefile]: I've dropped `xml.pom`, so now it's only in two places.

**Review by mandrigin — APPROVED** [review]
lgtm

**Review by pedropombeiro — APPROVED** [review]

**Review by cammellos — APPROVED** [review]

**Review by janherich — APPROVED** [review]

**Review by rasom — APPROVED** [review]

**Comment by lukaszfryc** [comment]
@jakubgs I cannot run macOS build (I tried fresh install and upgrade scenarios). Could you check the screenshot I'm attaching? Other than that the PR looks good.

image

**Comment by jakubgs** [comment]
Thanks @lukaszfryc but that makes no sense to me. This PR changes nothing about desktop builds. It only affects mobile builds. Can you link to me which build artifact you used?

**Comment by lukaszfryc** [comment]
I used https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181003-191815-a04d8e-pr.dmg from https://github.com/status-im/status-react/pull/5888#issuecomment-426771813

<blockquote></blockquote>

**Comment by lukaszfryc** [comment]
@jakubgs now all platforms work fine!

**Pull request merged**